### ISSUE TITLE
Support for resolutionPolicy on azurerm_private_dns_zone_virtual_network_link

### ISSUE THREAD

**Issue by ineednewpajamas** [hashicorp/terraform-provider-azurerm#28087]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave comments along the lines of "+1", "me too" or "any updates", they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment and review the [contribution guide](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/README.md) to help.

### Description

Please add `resolution_policy` to`azurerm_private_dns_zone_virtual_network_link`.
The [ResolutionPolicy](https://learn.microsoft.com/en-us/java/api/com.azure.resourcemanager.privatedns.models.resolutionpolicy) property is available in API version 2024-06-01 or higher.
Supported values are `Default` or `NxDomainRedirect`

### New or Affected Resource(s)/Data Source(s)

azurerm_private_dns_zone_virtual_network_link

### Potential Terraform Configuration

```hcl
resource "azurerm_private_dns_zone_virtual_network_link" "example" {
  name                  = "test"
  resource_group_name   = azurerm_resource_group.example.name
  private_dns_zone_name = azurerm_private_dns_zone.example.name
  virtual_network_id    = azurerm_virtual_network.example.id
  resolution_policy     = "NxDomainRedirect"
}
```

### References

https://learn.microsoft.com/en-us/azure/dns/private-dns-fallback

**Comment by anwarnk** [comment]
This would be great if it could be implemented, we have waiting for something this for a long time.

**Comment by pelikanmartin** [comment]
Same here! A long-awaited feature from Microsoft for many.

**Comment by pkoseksmt** [comment]
Same here!! Thanks

**Comment by ineednewpajamas** [comment]
While increasing the API is the first step, it does not resolve this issue. The resolution_policy  attribute still needs to be exposed

**Comment by EppO** [comment]
indeed, #28599 just a small piece of the puzzle, implementing the new attribute is still needed. I'll work on a new PR to implement that.

**Comment by EppO** [comment]
#28615 is ready for review and will implement the new setting
